### PR TITLE
:sparkles: Adding Analyzer and Maven Agents

### PR DESCRIPTION
**Pull request by shawn-hurley**
also in this commit is the following:
* Remove xml line number parser and moved to lxml for xml
* Changed the way dep matches and overriding of deps was handled To not rely on line numbers
* Changed AgentResult to have defaults and updated a field from a list to a single value. This is because we should only work on a single file at a time, and we don't need more than one file in the agent.
* Added a test case for the AnalyzerAgent

### CODE REVIEW

File: kai_solution_server/service/incident_store/incident_store.py [hunk 1]
```diff
@@ -528,6 +528,9 @@ def incident_store_from_config(config: KaiConfig) -> "IncidentStore":
         KAI_LOG.info(f"Selected provider: {config.models.provider}")
         KAI_LOG.info(f"Selected model: {model_provider.model_id}")
 
+        if config.incident_store is None:
+            raise ValueError
+
```
Comment (JonahSussman) [inline kai_solution_server/service/incident_store/incident_store.py:533]: This currently raises a `type[ValueError]` not a `ValueError`. I think you want to raise `ValueError()` instead.

File: requirements.txt
Comment (JonahSussman) [inline requirements.txt:1]: I'm just commenting for discussion, but is there a benefit of having this file anymore? All the requirements we have are in `pyproject.toml` now. 

@fabianvf @pranavgaikwad thoughts?

**Review by JonahSussman — CHANGES_REQUESTED** [review]
One potential bug, spelling fix/nits. Otherwise looks good!

File: kai/reactive_codeplanner/agent/analyzer_fix/agent.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        """Private method that will be used to parse the contents and get the results"""
+
+        lines_of_output = cast(str, message.content).splitlines()
+
+        in_java_file = False
+        in_reasoning = False
+        in_additional_details = False
+        java_file = ""
+        reasoning = ""
+        additional_details = ""
```
[the remark this answers is not in the document]
Comment (shawn-hurley) [reply kai/reactive_codeplanner/agent/analyzer_fix/agent.py:151]: Both none and empty strings will give you false values, right? If that is the case, I think this is fine. Thoughts?

**Pull request closed without merging**